### DISCUSSION TITLE
Issues with Using getReport2 for Data Extraction via Panel's JSON Template

### DISCUSSION THREAD

**Discussion by 0JaeminKim0** [pitchmuc/adobe_analytics_api_2.0#103] [Q&A]
Hello,

I am currently utilizing the getReport2 function to extract data using Panel's JSON template. The data extraction process involves retrieving information across four dimensions, with each dimension comprising 5 to 6 features, and the DateRange spans approximately 7 weeks.

However, I am encountering three issues *irregularly during the execution:
* There are no regular error iter while we running 76 iter wit for loop.

1. Key Error 'columns' (aanalytics2:3001)
2. Key Error 'name' (aanalytics2:3001)
3. AttributeError: 'Response' object has no attribute 'get' — This indicates that the response has not been received at connectors_170 line.

I would greatly appreciate any assistance or guidance on resolving these issues.

Thank you.

**Comment by pitchmuc** [comment]
Hey @0JaeminKim0 , 

I think it is because you are hitting the thottle limit of the Analytics API and some reports are not getting a full formatted JSON. 
So it is erroring out while trying to use the JSON, I should probably check the response of the JSON before using it ... but I am not sure to expect from your requests. The module has a single point for using POST method...
Can you try to do slow down the generation of these reports (adding a sleep) and see if it works ? 
Otherwise, I would try to do them one by one and try/catch them see which one fail, they may have something that I am not expected in their JSON. 

(However, 'Response' object has no attribute 'get' is exactly as you described, the server did not respond, so I bet you are going faster than what the Analytics API can handle - it is 12 request for 6 seconds)

**Reply by 0JaeminKim0** [reply]
Thank you for answering.

I have solved(?) with using try and exception with recursive function to call getreport2 function :)

Many thanks
